src/Term/TermFallback.php

```diff
@@ -46,6 +47,28 @@ public function __construct( $requestedLanguageCode, $text, $actualLanguageCode,
 			throw new InvalidArgumentException( '$sourceLanguageCode must be a non-empty string or null' );
 		}
 
+		if ( $sourceLanguageCode === null && $actualLanguageCode === $requestedLanguageCode ) {
```

Comment:
As I said, what's relevant here are the tests. If they cover everything, anybody can safely remove/refactor these checks.